### ISSUE TITLE
KenLM or similar language model

### ISSUE THREAD

**Issue by jesusbft** [mittagessen/kraken#682]
Hello,

Do Kraken plan to integrate KenLM or similar language model to help fixing errors in transcriptions? If yes, already have something (codes and ideas) that I could use? PyLaia have something but I never test it.

Thank you,
Weslley O.

**Comment by mittagessen** [comment]
There is already a beam decoder in `lib/ctc_decoder.py` which is
compatible with KenLM but proper language model integration is somewhat
tricky. The issue is that decoding happens in label space but the
language model works code point space (in whatever granularity it's been
trained in). This makes it quite easy to create broken combinations.

The new party recognizer on the other hand has much stronger language
modeling with its pretrained Llama decoder so there shouldn't be any
need for an external LM anymore. It is going to end up in kraken as soon
as I figure out how to fine-tune the thing (and got the time to start
the integration work).

**Comment by jesusbft** [comment]
Thank you so much for the informations. I'll take a look in both!!
